cmd/kubeadm/app/cmd/token.go

```diff
@@ -0,0 +1,183 @@
+			err := RunListTokens(out, tokenCmd)
+			kubeadmutil.CheckErr(err)
+		},
+	}
+	tokenCmd.AddCommand(listCmd)
+
+	deleteCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete discovery tokens on the server.",
+		Run: func(tokenCmd *cobra.Command, args []string) {
+			err := RunDeleteToken(out, tokenCmd, args[0])
+			kubeadmutil.CheckErr(err)
+		},
+	}
+	tokenCmd.AddCommand(deleteCmd)
+
+	return tokenCmd
+}
+
+// RunCreateToken generates a new bootstrap token and stores it as a secret on the server.
+func RunCreateToken(out io.Writer, cmd *cobra.Command, tokenDuration time.Duration, tokenSecret *kubeadmapi.Secrets) error {
+	client, err := kubemaster.CreateClientFromFile(path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "admin.conf"))
+	if err != nil {
+		return err
+	}
+
+	err = kubeadmutil.GenerateTokenIfNeeded(tokenSecret)
+	if err != nil {
+		return err
+	}
+
+	secret := &api.Secret{
+		ObjectMeta: api.ObjectMeta{
+			Name: fmt.Sprintf("%s%s", bootstrapTokenSecretPrefix, tokenSecret.TokenID),
+		},
+		Type: api.SecretTypeBootstrapToken,
+		Data: encodeTokenSecretData(tokenSecret, tokenDuration),
+	}
+	if _, err := client.Secrets(api.NamespaceSystem).Create(secret); err != nil {
+		return fmt.Errorf("<cmd/token> failed to bootstrap token [%v]", err)
+	}
+	fmt.Println(tokenSecret.GivenToken)
+
+	return nil
+}
+
+func encodeTokenSecretData(tokenSecret *kubeadmapi.Secrets, duration time.Duration) map[string][]byte {
+	var (
+		data = map[string][]byte{}
+	)
+
+	data["token-id"] = []byte(tokenSecret.TokenID)
+	data["token-secret"] = []byte(tokenSecret.BearerToken)
+
+	t := time.Now()
+	t = t.Add(duration)
+	data["expiration"] = []byte(t.Format(time.RFC3339))
+
+	return data
+}
+
+// RunListTokens lists details on all existing bootstrap tokens on the server.
+func RunListTokens(out io.Writer, cmd *cobra.Command) error {
+	client, err := kubemaster.CreateClientFromFile(path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "admin.conf"))
+	if err != nil {
+		return err
+	}
+
+	tokenSelector := fields.SelectorFromSet(
+		map[string]string{
+			api.SecretTypeField: string(api.SecretTypeBootstrapToken),
+		},
+	)
+	listOptions := api.ListOptions{
+		FieldSelector: tokenSelector,
+	}
+
+	results, err := client.Secrets(api.NamespaceSystem).List(listOptions)
+	if err != nil {
+		return fmt.Errorf("<cmd/token> failed to list bootstrap tokens [%v]", err)
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 10, 4, 3, ' ', 0)
+	fmt.Fprintln(w, "ID\tTOKEN\tEXPIRATION")
+	for _, secret := range results.Items {
+		tokenId := secret.Data["token-id"]
+		token := fmt.Sprintf("%s.%s", tokenId, secret.Data["token-secret"])
+		expireTime, err := time.Parse(time.RFC3339, string(secret.Data["expiration"]))
+		if err != nil {
+			return fmt.Errorf("<cmd/token> error parsing expiry time [%v]", err)
+		}
+		expires := kubectl.ShortHumanDuration(expireTime.Sub(time.Now()))
+		fmt.Fprintf(w, "%s\t%s\t%s\n", tokenId, token, expires)
+	}
+	w.Flush()
+
+	return nil
```

Comment:
Eventually folks are going to ask for ways to script this and we'll find ourselves reimplementing all of the output modes that `kubectl` has.  This is good for now but won't last.

Also -- most users see the key as `123456.1234567890123456`.  I'm not sure we need to break it down to id and token (I called that part "secret").  I'd just name the column "token" and have the dotted form.